src/overlays/actors/ovl_En_Mkk/z_en_mkk.c

```diff
@@ -82,62 +111,500 @@ static DamageTable D_80A4F754 = {
+    if (this->collider.base.atFlags & AT_ON) {
+        CollisionCheck_SetAT(play, &play->colChkCtx, &this->collider.base);
+    }
+    if (this->collider.base.acFlags & AT_ON) {
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.base);
+    }
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+}
+
+void func_80A4EDF0(EnMkk* this) {
+    this->alpha = 0;
+    this->unk_14B |= 3;
+    this->actor.flags &= ~ACTOR_FLAG_1;
+    this->actor.draw = NULL;
+    this->actor.update = func_80A4F16C;
+    this->actor.gravity = -0.5f;
+    this->unk_14E = this->unk_152;
+    this->actionFunc = func_80A4EE48;
+}
+
+void func_80A4EE48(EnMkk* this, PlayState* play) {
+    if (this->unk_14E > 0) {
+        this->unk_14E--;
+        if (this->unk_14E == 0) {
+            this->actor.flags &= ~ACTOR_FLAG_10;
+        }
+    } else if ((this->actor.xzDistToPlayer < this->unk_178) && (Player_GetMask(play) != PLAYER_MASK_STONE)) {
+        this->actor.update = EnMkk_Update;
+        this->actor.draw = EnMkk_Draw;
+        this->actor.flags &= ~ACTOR_FLAG_10;
+        func_80A4E0CC(this);
+    }
+}
+
+void func_80A4EEF4(EnMkk* this) {
+    Math_Vec3f_Copy(&this->unk_16C, &this->actor.world.pos);
+    Math_Vec3f_Copy(&this->unk_154, &this->actor.world.pos);
+    Math_Vec3f_Copy(&this->unk_160, &this->actor.world.pos);
+    Actor_SetScale(&this->actor, 0.005f);
+    this->actor.update = func_80A4F16C;
+    this->actor.draw = func_80A4F4C8;
+    this->actionFunc = func_80A4EF74;
+}
+
+void func_80A4EF74(EnMkk* this, PlayState* play) {
+    s32 newAlpha = this->alpha - 20;
+
+    if (newAlpha <= 0) {
+        if (this->unk_14B & 8) {
+            Actor_SetScale(&this->actor, 0.01f);
+            this->primColorSelect = 0;
+            this->unk_149 = 0;
+            Math_Vec3f_Copy(&this->actor.world.pos, &this->actor.home.pos);
+            Math_Vec3f_Copy(&this->unk_154, &this->actor.world.pos);
+            Math_Vec3f_Copy(&this->unk_160, &this->actor.world.pos);
+            this->actor.speedXZ = 0.0f;
+            this->actor.velocity.y = 0.0f;
+            func_80A4EDF0(this);
+        } else {
+            Actor_MarkForDeath(&this->actor);
+        }
+    } else {
+        this->alpha = newAlpha;
+        this->actor.world.pos.x += 0.3f + (3.0f * Rand_ZeroOne());
+        this->actor.world.pos.y += 0.5f + (3.5f * Rand_ZeroOne());
+        this->unk_16C.x -= 0.3f + (3.0f * Rand_ZeroOne());
+        this->unk_16C.y += 0.5f + (3.5f * Rand_ZeroOne());
+        this->unk_154.z += 0.3f + (3.0f * Rand_ZeroOne());
+        this->unk_154.y += 0.5f + (3.5f * Rand_ZeroOne());
+        this->unk_160.z -= 0.3f + (3.0f * Rand_ZeroOne());
+        this->unk_160.y += 0.5f + (3.5f * Rand_ZeroOne());
+    }
+}
+
+void func_80A4F16C(Actor* thisx, PlayState* play) {
+    EnMkk* this = THIS;
+
+    this->actionFunc(this, play);
+}
+
+void EnMkk_Draw(Actor* thisx, PlayState* play) {
+    struct EnMkkDlists* dLists;
```

Comment:
```suggestion
    struct EnMkkDlists* dLists = &sBoeDLists[this->actor.params];
```
Does this work? If not could you move things around to get it to work?

Reply:
mmm no, i already tried (again), the best i can offer you is
EnMkkDlists* dLists = &sBoeDLists[thisx->params];
otherwise it messes up the stack